client/scripts/views/pages/new_thread.ts

```diff
@@ -135,14 +144,22 @@ export const NewThreadForm: m.Component<{}, IState> = {
         ]),
         m(FormGroup, [
           m(Button, {
+            class: !author ? 'disabled' : '',
             intent: 'primary',
             label: 'Create link',
+            name: 'submission',
             onclick: () => {
               if (!vnode.state.error.url && !detectURL(vnode.state.form.url)) {
                 vnode.state.error.url = 'Must provide a valid URL.';
               }
               if (!Object.values(vnode.state.error).length) {
-                Object.assign(vnode.state.error, newLink(vnode.state.form, vnode.state.quillEditorState, author));
+                newLink(vnode.state.form, vnode.state.quillEditorState, author);
+              }
+              if (!vnode.state.error) {
+                $(vnode.dom).trigger('modalcomplete');
+                setTimeout(() => {
```

Comment:
why not just call the modal exit directly?

Reply:
I was following the pattern in input_modal.ts and confirm_modal.ts

Now that I do a formal survey, it seems like we have three styles:
1. The 'modalcomplete' + timeout'd 'modalexit'
2. The 'modalcomplete' on its own
3. A non-timeout'd 'modalexit'

How do folks feel about these?
Examples for context
```
$(vnode.dom).trigger('modalcomplete');
setTimeout(() => {
          $(vnode.dom).trigger('modalexit');
    }, 0);

$(vnode.dom).trigger('modalcomplete');

$(vnode.dom).trigger('modalexit');
```

Reply:
The modalcomplete event should only be called if whatever created the modal set up a callback. Otherwise it can be left out.

I added a clarification about this to the modal documentation in client/scripts/views/modal.ts :-)